SU-HSDO/suhumsci
# STN-783: Vertical Timeline Triangle Icon Expand/Collapse Indicator Bug

# READY FOR REVIEW

## Summary
The work in this PR solves an issue on Vertical Timeline components where the triangle icon expand/collapse indicator did not flip up and down vertically when a timeline item was clicked upon. While debugging the issue I discovered the aria attributes set on timeline item summaries were not updating on click as they should. The triangle icon styles target the aria attribute values to determine whether the triangle icon should point up or point down.

By resolving aria attributes issue, the triangle icon display as expected.

## Need Review By (Date)
6/11

## Urgency
medium

## Steps to Test
1. In your CLI, run `npm run test` to confirm all tests pass.
2. Clear your cache, then run `npm run build` to build the JS in this PR.
3. If you have a page with vertical timeline components, you may use that to test the work in this PR. If not, create a flexible page with a vertical timeline that has timeline items collapsed and a vertical timeline with timeline items expanded.
4. - [x] Confirm that when you click on an items within both the expanded and collapsed timelines, the triangle icon flips. When a timeline item is collapsed the triangle icon should point down. When a timeline item is expanded the triangle should point up. Note: It may be easier to test this in a logged out state or an incognito window to avoid the Drupal contextual links which overlay the triangle icons when logged in.
5. - [x] Confirm that you can navigate through timeline items using a keyboard only and that the triangle icon flips when expanding/collapsing items.
6. Open your browser's developer tools and test the following:
    - [x] In the console tab, confirm there are no JS errors related to the timeline.
    - [x] In the elements tab, locate the aria-expanded and aria-pressed attributes on timeline item summaries and confirm that they update on click and when navigating via the keyboard.
    <img width="753" alt="aria attributes" src="https://user-images.githubusercontent.com/12678977/121391334-3f1c0400-c91c-11eb-80d4-60c7b7745e18.png">
